libmamba/src/core/pool.cpp

```diff
@@ -12,22 +12,26 @@ namespace mamba
 {
     void libsolv_debug_callback(Pool* pool, void* userptr, int type, const char* str)
     {
-        auto* logger = (spdlog::logger*)(userptr);
-        std::string s(str);
-        if (strip(s).size() == 0) return;
-        if (s.back() != '\n') s.push_back('\n');
+        auto* dbg = (std::pair<spdlog::logger*, std::string>*)(userptr);
+        dbg->second += str;
+        if (dbg->second.size() == 0 || dbg->second.back() != '\n')
+        {
+            return;
+        }
+
         if (type & SOLV_FATAL || type & SOLV_ERROR)
         {
-            logger->error(s);
+            dbg->first->error(dbg->second);
         }
         else if (type & SOLV_WARN)
         {
-            logger->warn(s);
+            dbg->first->warn(dbg->second);
         }
         else if (Context::instance().verbosity > 2)
         {
-            logger->info(s);
+            dbg->first->info(dbg->second);
         }
+        dbg->second.clear();
```

Comment:
@wolf what was the idea behind the `std::string` callback data member? it is cleared afterwards and never used again. Was it for reusing the alloc?
Wouldn't wrapping in a `std::string_view` suffice here?

Reply:
Only if we don't return before (line 17)

Reply:
But I am sure this can be improved!